charmcraft/commands/store/store.py

```diff
@@ -210,10 +212,20 @@ def whoami(self):
         """Return authenticated user details."""
         response = self._client.whoami()
 
-        result = User(
-            name=response["display-name"],
-            username=response["username"],
-            userid=response["id"],
+        acc = response["account"]
+        account = Account(name=acc["display-name"], username=acc["username"], id=acc["id"])
+        if response["packages"] is None:
+            packages = None
+        else:
+            packages = [
+                Package(type=pkg["type"], name=pkg.get("name"), id=pkg.get("id"))
+                for pkg in response["packages"]
+            ]
+        result = MacaroonInfo(
+            account=account,
+            packages=packages,
+            channels=response["channels"],
+            permissions=response["permissions"],
```

Comment:
is it possible for charmhub to not return these keys at all

Reply:
Nop, check the [API Spec for the used endpoint](https://api.charmhub.io/docs/default.html#macaroon_info):

```
    "required": [
        "account",
        "channels",
        "packages",
        "permissions"
    ],
```

They may be `null`, though, which is supported in the code...